weldx/core.py

```diff
@@ -0,0 +1,341 @@
+                self.interp_time(Q_([1, 2], "second"))
+                self.interp_time(Q_([1, 2, 3], "second"))
+            except Exception as e:
+                raise Exception(
+                    "The expression can not be evaluated with arrays of time deltas. "
+                    "Ensure that all parameters that are multiplied with the time "
+                    "variable have an outer dimension of size 1. This dimension is "
+                    "broadcasted during multiplication. The original error message was:"
+                    f' "{str(e)}"'
+                )
+
+        else:
+            raise TypeError(f'The data type "{type(data)}" is not supported.')
+
+    @property
+    def data(self) -> Union[xr.DataArray, MathematicalExpression]:
+        """Return the data of the TimeSeries.
+
+        This is either a set of discrete values or a mathematical expression.
+
+        Returns
+        -------
+        xr.DataArray:
+            Underlying data array with discrete values in time.
+        MathematicalExpression:
+            A mathematical expression describing the time dependency
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            return self._data.data
+        return self._data
+
+    @property
+    def interpolation(self) -> Union[str, None]:
+        """Return the interpolation.
+
+        Returns
+        -------
+        str:
+            Interpolation of the TimeSeries
+
+        """
+        return self._interpolation
+
+    @property
+    def time(self) -> Union[None, pd.TimedeltaIndex]:
+        """Return the data's timestamps.
+
+        Returns
+        -------
+        pd.TimedeltaIndex:
+            Timestamps of the  data
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            if len(self._data.time) == 1:
+                return None
+            return ut.to_pandas_time_index(self._data.time.data)
+        return self._time
+
+    def interp_time(self, time: Union[pd.TimedeltaIndex, Q_]):
+        """Interpolate the TimeSeries in time.
+
+        If the internal data consists of discrete values, an interpolation with the
+        prescribed interpolation method is performed. In case of mathematical
+        expression, the expression is evaluated for the given timestamps.
+
+        Parameters
+        ----------
+        time:
+            A set of timestamps.
+
+        Returns
+        -------
+        xr.DataArray:
+            A data array containing the interpolated data.
+
+        """
+        if isinstance(self._data, xr.DataArray):
+            if self._interpolation == "linear":
+                return ut.xr_interp_like(
+                    self._data,
+                    {"time": time},
+                    assume_sorted=False,
+                    broadcast_missing=False,
+                )
+            raise Exception("not implemented")
+
+        if not isinstance(time, Q_) or not time.check(UREG.get_dimensionality("s")):
```

Comment:
I would prefer using `pint.Quantity` over `Q_` for `isinstance()` and all type hints

Reply:
Done